docs/gcloud-auth.rst

```diff
@@ -0,0 +1,264 @@
+You should simply write your code as though you've already authenticated
+as we can discover your credentials and Project ID automatically.
+
+The following code should "just work" inside GAE or GCE:
+
+.. code-block:: python
+
+   from gcloud import pubsub
+   pubsub.Topic('topic_name').create()
+
+---------------------------------------
+Using a different key inside GCE or GAE
+---------------------------------------
+
+You might be running inside GCE or GAE but want to
+use a different Service Account.
+In that case, jump up to the section about
+using a specific keyfile.
+Thanks to the order of evaluation,
+the keyfile you specify explicitly or via the environment variable
+will take precedence over the automatically discovered key in the
+App Engine or Compute Engine environment.
+
+=============================================
+Using multiple credentials in the same script
+=============================================
+
+There may be times where you want to use
+multiple different sets of credentials inside the same script.
+To do this, you should create miltiple connections
+and specify which to use on the various API calls you make.
+
+For example, here is how you would create two Pub/Sub topics
+in two different projects
+with two different sets of credentials.
+
+.. code-block:: python
+
+   from gcloud import pubsub
+   from gcloud.credentials import get_for_service_account_json
+
+   # Create two different credentials.
+   credentials1 = get_for_service_account_json('key1.json')
+   credentials2 = get_for_service_account_json('key2.json')
+
+   # Create two different connections.
+   connection1 = pubsub.Connection(credentials=credentials1)
+   connection2 = pubsub.Connection(credentials=credentials2)
+
+   # Create two different topics.
+   pubsub.Topic('topic1', project='project1', connection=connection1).create()
+   pubsub.Topic('topic2', project='project2', connection=connection2).create()
+
+If you have one "main" set of credentials
+and others are "one-offs" for special cases,
+you can avoid all this typing
+by using the default connection for the main set,
+and separate connections for the others.
+
+Using the same example as before,
+and assuming ``'key1.json'`` is the "main" set of credentials,
+here's what that same code might look like.
+
+.. code-block:: python
+
+   from gcloud import pubsub
+   from gcloud.credentials import get_for_service_account_json
+
+   credentials2 = get_for_service_account_json('key2.json')
+   connection2 = pubsub.Connection(credentials=credentials2)
+
+   pubsub.Topic('topic1').create()
+   pubsub.Topic('topic2', project='project2', connection=connection2).create()
+
+===============
+Troubleshooting
+===============
+
+--------------------------------------------------
+You keep mentioning key.json... How do I get that?
+--------------------------------------------------
+
+When we say ``key.json``
+we're referring to what we call a "Service Account".
+These are sort of like passwords
```

Comment:
I think I may have implied that people using Google Cloud are the same people who have configured a MySQL Database. I meant "turned on".

We also have to consider the scenario where people are coming from App Engine which requires **zero** authentication to talk to Datastore...

I think it's worthwhile to clarify what the thing is with one sentence. If you want to shorten it up or correct it, we should do that. It's my feeling that if someone can't figure out what `key.json` is on their own, they need hand holding and probably don't even know what a service accounts (or credentials JSON files) are...